Document/Testcases/0x00b_OMTG-DATAST_iOS.md

```diff
@@ -1,5 +1,54 @@
-## <a name="OMTG-DATAST-002"></a>OMTG-DATAST-002: Testing for Sensitive Data Disclosure in Log Files
+## <a name="OMTG-DATAST-001-1"></a>OMTG-DATAST-001-1: Test for system credentials storage features 
+
+### Black-box Testing
+
+A way to identify if sensitive information like credentials and keys are stored insecurely and without leveraging the native functions from IOS is to analyse the app data directory. It is important to trigger as much app functionality as possbile before the data is analysed, as the app might only store system credentials as specific functionality is triggered by the user. A static analysis can then be performed for the data dump based on generic keywords and app specifc data. Identify how the application stores data locally on the iOS device. Some of the possible options for the application to store it's data locally includes:
+ 
+- Plain Files
+- Plist
+- SQLite3 DB
+- Realm DB
+
+Steps :
+
+1. Proceed to trigger functionality that stores potential sensitive data.
+2. SSH into your iOS device and browse to the following directory: `/var/mobile/Containers/Data/Application/$APP_ID/`
+3. Perform a grep command of the data that you have stored, such as: `grep -irn "jk@vantagepoint.sg"`.
+4. If the sensitive data is being stored in plaintext, it fails this test.
+
+Manual dynamic analysis such as debugging can also be leveraged to verify how specific system credentials are stored and processed on the device. As this approach is more time consuming and is likely conducted manually, it might be only feasible for specific use cases.  
+
+
+### White-box Testing
+When going through the source code it should be analyzed if native mechanisms that are offered by IOS are applied to the identified sensitive information. Ideally sensitive information should not be stored on the device at all. If there is a requirement to store sensitive information on the device itself, several functions/API calls are available to protect the data on IOS device by using for example the Keychain. 
```

Comment:
Be more specific. What native mechanisms are there? What are the functions call I need to look for? What are the best practices? Are there differences in different iOS versions?